### PR TITLE
Add isRotationSupported() and refine SPOSet.cpp error messages

### PR DESCRIPTION
**Pull request by ye-luo**
## Proposed changes
Better safeguards targeting internal logic consistency.

## What type(s) of changes does this code introduce?
- New feature

### Does this introduce a breaking change?
- No

## What systems has this change been tested on?
epyc

## Checklist
- Yes. This PR is up to date with current the current state of 'develop'

### CODE REVIEW

File: src/QMCWaveFunctions/SPOSetBuilder.cpp [hunk 1]
```diff
@@ -98,6 +98,9 @@ std::unique_ptr<SPOSet> SPOSetBuilder::createSPOSet(xmlNodePtr cur)
     // create sposet with rotation
     auto& sposet_ref = *sposet;
     app_log() << "  SPOSet " << sposet_ref.getName() << " is optimizable\n";
+    if (!sposet_ref.isRotationSupported())
```
Comment (PDoakORNL) [inline src/QMCWaveFunctions/SPOSetBuilder.cpp:101]: Why not throw a UniformCommunicateError here?

**Review by PDoakORNL — CHANGES_REQUESTED** [review]
This looks fine to me except for the SPOSetBuilder error.

File: src/QMCWaveFunctions/SPOSetBuilder.cpp (hunk 1 above)
Comment (ye-luo) [reply src/QMCWaveFunctions/SPOSetBuilder.cpp:101]: At this point, I can already safely stop the code and mpi communicator is available here. I think UniformCommunicateError is better suited in classes without mpi communicator.
returning UniformCommunicateError may got stuck if another exception happens in destruction due to earlier issues. So I feel better to stop early whenever possible.

Comment (PDoakORNL) [reply src/QMCWaveFunctions/SPOSetBuilder.cpp:101]: Seems reasonable.

**Review by PDoakORNL — APPROVED** [review]

**Comment by ye-luo** [comment]
Test this please

**Pull request merged**